pkg/collector/corechecks/cluster/orchestrator/processors/processor.go

```diff
@@ -180,17 +232,19 @@ func (p *Processor) Process(ctx *ProcessorContext, list interface{}) (processRes
 
 	processResult = ProcessResult{
 		MetadataMessages: ChunkMetadata(ctx, p, resourceMetadataModels, resourceManifestModels),
-		ManifestMessages: ChunkManifest(ctx, p.h.BuildManifestMessageBody, resourceManifestModels),
+	}
+	if ctx.IsManifestProducer() {
+		processResult.ManifestMessages = ChunkManifest(ctx, p.h.BuildManifestMessageBody, resourceManifestModels)
 	}
 
 	return processResult, len(resourceMetadataModels)
 }
 
 // ChunkManifest is to chunk Manifest payloads
-func ChunkManifest(ctx *ProcessorContext, buildManifestBody func(ctx *ProcessorContext, resourceManifests []interface{}, groupSize int) model.MessageBody, resourceManifestModels []interface{}) []model.MessageBody {
+func ChunkManifest(ctx ProcessorContext, buildManifestBody func(ctx ProcessorContext, resourceManifests []interface{}, groupSize int) model.MessageBody, resourceManifestModels []interface{}) []model.MessageBody {
 	// Chunking resources based on the serialized size of their manifest and maximum messages number
 	// Chunk manifest messages and use itself as weight indicator
-	chunks := chunkOrchestratorPayloadsBySizeAndWeight(resourceManifestModels, resourceManifestModels, ctx.Cfg.MaxPerMessage, ctx.Cfg.MaxWeightPerMessageBytes)
+	chunks := orchestrator.ChunkOrchestratorPayloadsBySizeAndWeight(resourceManifestModels, resourceManifestModels, weightForOrchestratorPayload, ctx.GetOrchestratorConfig().MaxPerMessage, ctx.GetOrchestratorConfig().MaxWeightPerMessageBytes)
```

Comment:
Do we plan on having a different `weightForOrchestratorPayload` function in the future? I'm mostly asking because we seem to have extracted the functionality to have it being different, but I do not see in this PR using a different function.

Reply:
In the new PR https://github.com/DataDog/datadog-agent/pull/22060 I reverted this part